lib/fastly_nsq/listener.rb

```diff
@@ -0,0 +1,59 @@
+require 'fastly_nsq/message'
+
+module FastlyNsq
+  class Listener
+    def self.listen_to(**args)
+      new(**args).go
+    end
+
+    def initialize(topic:, processor:, channel: nil, consumer: nil, **options)
+      @topic        = topic
+      @processor    = processor
+      @consumer     = consumer || FastlyNsq::Consumer.new(topic: topic, channel: channel)
+      @logger       = options.fetch :logger, FastlyNsq.logger
+      @preprocessor = options[:preprocessor]
+    end
+
+    def go(run_once: false)
```

Comment:
Discussed this IRL: refactoring makes this method more clear; `go` could just be called `run` with a companion method `run_once`, even if the `run_once` method were only used in tests, it's a small difference compared to passing in as a parameter. For now, leaving with one `go` method (and parameter) but could be something to refactor in a future PR.